### PR TITLE
Add more @showarg examples

### PR DESCRIPTION
**Pull request by bluesmoon**
Adds a few more examples to show how @showarg works with literals and interpolated string expressions.

### CODE REVIEW

**Review by fingolfin — APPROVED** [review]
I find the selection of examples a bit arbitrary but if you find it helpful, why not :-)

**Comment by vtjnash** [comment]
The docstring REPL does not permit these sort of inline comments. I am not sure this is that useful, since it is also just showing that different expressions parse differently, which is probably explained elsewhere (though I agree that it can take a while to fully appreciate the implications of that)

File: doc/src/manual/metaprogramming.md [hunk 1]
```diff
@@ -629,6 +629,18 @@ julia> @showarg(1+1)
 
 julia> @showarg(println("Yo!"))
 :(println("Yo!"))
+
+# Numeric literal
+julia> @showarg(1)
```
Comment (fingolfin) [inline doc/src/manual/metaprogramming.md:634]: You'd have to do it like this:
```suggestion
julia> @showarg(1)   # Numeric literal
```

**Comment by fingolfin** [comment]
@vtjnash that's a valid argument against merging this, of course.

So who makes the decision whether to reject this, or whether to ask for further improvements? I personally could live with either, but we should not leave @bluesmoon hanging

**Comment by bluesmoon** [comment]
I'll update the MR.

**Comment by bluesmoon** [comment]
I've updated the MR with @vtjnash's suggestion.

**Pull request merged**